File: Programmers/Lv_1/[1차]비밀지도.js
function solution(n, arr1, arr2) {
    return arr1.map((v, i) => addZero(n, (v | arr2[i]).toString(2)).replace(/1|0/g, a => +a ? '#' : ' '));
}

const addZero = (n, s) => {
    return '0'.repeat(n - s.length) + s;
}

var solution = (n, a, b) => a.map((a, i) => (a | b[i]).toString(2).padStart(n, 0).replace(/0/g, ' ').replace(/1/g, '#'));

function solution(n, arr1, arr2) {
    let num1, num2, s;
    let answer = [];

    for (let i = 0; i < n; i++) {
        num1 = arr1[i];
        num2 = arr2[i];
        s = '';

        for (let j = 0; j < n; j++) {
            s = (num1 % 2 + num2 % 2) ? '#'+s : ' ' + s;
            num1 = Math.floor(num1 / 2);
            num2 = Math.floor(num2 / 2);
        }
        answer.push(s);
    }
    return answer;
}

File: Programmers/Lv_1/[1차]비밀지도.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string to_binary(int size, int n) {
    string result = "";
    while (n > 0) {
        result += to_string(n % 2);
        n /= 2;
    }
    while (result.size() < size) {
        result += "0";
    }
    reverse(result.begin(), result.end());
    return result;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    string s;
    vector<string> s_arr1, s_arr2;

    for (int i = 0; i < n; i++) {
        s_arr1.push_back(to_binary(n, arr1[i]));
        s_arr2.push_back(to_binary(n, arr2[i]));
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (s_arr1[i][j] == '1' || s_arr2[i][j] == '1') s += "#";
            else s += " ";
        }
        answer.push_back(s);
        s = "";
    }

    return answer;
}

vector<string> best_solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;

    for (int i = 0; i < n; i++) {
        arr1[i] = arr1[i] | arr2[i];
        string ans = "";

        for (int j = 0; j < n; j++) {
            if (arr1[i] % 2 == 0) ans = " " + ans;
            else ans = "#" + ans;
            
            arr1[i] = arr1[i] >> 1;
        }
        answer.push_back(ans);
    }
}


int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int n, data;
    vector<int> arr1, arr2;
    vector<string> result;
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> data;
        arr1.push_back(data);
    }
    for (int i = 0; i < n; i++) {
        cin >> data;
        arr2.push_back(data);
    }

    result = solution(n, arr1, arr2);

    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << '\n';
    }

    return 0;
}